common/schema-builder/src/main/java/io/smallrye/graphql/schema/helper/RolesAllowedDirectivesHelper.java

```diff
@@ -0,0 +1,78 @@
+package io.smallrye.graphql.schema.helper;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+
+import io.smallrye.graphql.schema.Annotations;
+import io.smallrye.graphql.schema.model.DirectiveArgument;
+import io.smallrye.graphql.schema.model.DirectiveInstance;
+import io.smallrye.graphql.schema.model.DirectiveType;
+import io.smallrye.graphql.schema.model.Reference;
+import io.smallrye.graphql.schema.model.Scalars;
+
+public class RolesAllowedDirectivesHelper {
+
+    public final static DirectiveType ROLES_ALLOWED_DIRECTIVE_TYPE;
+
+    static {
+        ROLES_ALLOWED_DIRECTIVE_TYPE = new DirectiveType();
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setName("rolesAllowed");
+        Set<String> locations = new LinkedHashSet<>();
+        locations.add("FIELD_DEFINITION");
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setLocations(locations);
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setDescription("Used to specify the role required to execute a given field or operation.");
+        ROLES_ALLOWED_DIRECTIVE_TYPE.setRepeatable(false);
+
+        addArgument("value", Scalars.getStringScalar());
+    }
+
+    private static void addArgument(String name, Reference reference) {
+        DirectiveArgument arg = new DirectiveArgument();
+        arg.setName(name);
+        arg.setReference(reference);
+        ROLES_ALLOWED_DIRECTIVE_TYPE.addArgumentType(arg);
+    }
+
+    public List<DirectiveInstance> transformRolesAllowedToDirectives(Annotations methodAnnotations,
+            Annotations classAnnotations) {
+        List<DirectiveInstance> result = new ArrayList<>();
+
+        Set<DotName> annotationNames = new HashSet<>(methodAnnotations.getAnnotationNames());
+
+        if (classAnnotations != null) {
+            annotationNames.addAll(classAnnotations.getAnnotationNames());
+        }
+
+        for (DotName annotationName : annotationNames) {
+            if (annotationName.equals(DotName.createSimple("jakarta.annotation.security.RolesAllowed"))) {
+                DirectiveInstance directive = new DirectiveInstance();
+                directive.setType(ROLES_ALLOWED_DIRECTIVE_TYPE);
+                String value = getStringValue(methodAnnotations, annotationName, "value");
+                if (value == null) {
+                    value = getStringValue(classAnnotations, annotationName, "value");
+                }
+                directive.setValue("value", value);
+                result.add(directive);
+                break;
```

Comment:
really only the first one?

Reply:
> really only the first one?
Since the `@RolesAllowed` annotation is non-repeatable (afaik), I thought it would make sense to stop the iteration.

Reply:
But then it doesn't make much sense to return a List, does it?

Reply:
Hm, yea that's true. In the `Field` class there was no method for adding single `DirectiveInstance` (only for `List<DirectiveInstance>`). So I will just add it and refactor the code.